Fixes for Chromatic

allanlasser opened the pull request:
This makes changes to improve the stability of Chromatic renders across builds.

- It adds an interaction test to `RevisionsDialog` that does integration testing on the component. This can serve as a model for future component tests.
- Major change is updating `src/api/orgAndUser.js:getMe` to not check for a CSRF token when fetching user data, instead relying on a 404 response to know if the user is signed out.

We've also been granted a free open-source project plan by Chromatic, so this adds a "thank you" to the README at their request.

eyeseast commented:
Possibly unrelated to this, I get an error (`ReferenceError: getUpgradeUrl`) hitting the upgrade button on staging. Might be a missing import.

<img width="1506" alt="Screenshot 2024-01-16 at 11 07 55 AM" src="https://github.com/MuckRock/documentcloud-frontend/assets/25778/17a17463-bb50-427a-bf9a-3f90c675c0f3">

allanlasser commented:
> Possibly unrelated to this, I get an error (`ReferenceError: getUpgradeUrl`) hitting the upgrade button on staging. Might be a missing import.

Nice catch. The function was defined as `getUpgradeURL` and being called as `getUpgradeUrl`. I've renamed the function and fixed all references with `getUpgradeUrl`.
